feat: add time groupings to recents list

audrey-kho opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
Added time grouping text for:
- Today
- Yesterday
- Last 7 days
- Last 30 days
- Last 90 days
- Last 6 months
- Last 12 months


## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://jira.corp.adobe.com/browse/SITES-20148

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):
<img width="334" alt="Screenshot 2024-04-17 at 10 14 52 AM" src="https://github.com/adobe/aem-genai-assistant/assets/59587534/bfe5e7d4-f89d-4921-b6ae-bf1774f02c57">

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

vtsaplin reviewed (changes requested):
@audrey-kho Great! Let's keep the grouping logic as is, but I'd encapsulate the state and the code using some sort of memoization technique. What if we use a higher-order function for this?
Something like this `getGroupNameFromDate(date)`. The function should be defined outside of the component.

askayastha22 commented:
@audrey-kho Can you please look at why one of the tests is failing?

audrey-kho commented:
> @audrey-kho Can you please look at why one of the tests is failing?

@askayastha22 there was some confusion with the time grouping cutoffs. There were 2 ways to set the time groupings:
1. Include exact times in the Date object (Today = all times within 24 hours of the current time, so if now = May 01, 2024 14:45:00, then April 30, 2024 23:59:00 will be marked as 'Today')
2. Include only month, day, and year (Today = only May 1st)

The tests are checking for the first interpretation, while the actual code seems to be trying to do a mix. Which one do you think makes more sense?

askayastha22 commented:
> > @audrey-kho Can you please look at why one of the tests is failing?
> 
> @askayastha22 there was some confusion with the time grouping cutoffs. There were 2 ways to set the time groupings:
> 
> 1. Include exact times in the Date object (Today = all times within 24 hours of the current time, so if now = May 01, 2024 14:30:00, then April 30, 2024 23:59:00 will be marked as 'Today')
> 2. Include only month, day, and year (Today = only May 1st)
> 
> The tests are checking for the first interpretation, while the actual code seems to be trying to do a mix. Which one do you think makes more sense?

@audrey-kho The second interpretation makes more sense to me. In your example, seeing April 30, 2024 as 'Today' is confusing.

audrey-kho commented:
@askayastha22 I had the same opinion about the date cutoff so I already modified the code to fit the 2nd description. I just added a few more test cases to confirm that the correct label is being produced when the Date object is < 24 hours away from the current time. If the tests make sense, I think this should be ready for release 🥳

askayastha22 reviewed (approved):
LGTM

irenelagno commented on web-src/src/components/MainSidePanel.js:
```diff
@@ -190,12 +201,18 @@ export function MainSidePanel(props) {
+                const sessionDate = new Date(session.name);
+                const groupingLabel = groupingLabelGenerator(sessionDate);
+
+                return (<>
```
this will lead to an warning  in the dev console:
` index.js:1 Warning: Each child in a list should have a unique "key" prop.`

To prevent it, you need to add key on the element inside loop.  See https://medium.com/@ishaqibrahimbss/reacts-favorite-warning-each-child-in-a-list-should-have-a-unique-key-prop-909dfe50cbe3

I'll fix it in the scope of https://github.com/adobe/aem-genai-assistant/pull/273, no action needed for now
